addons/microsoft_calendar/utils/microsoft_calendar.py

```diff
@@ -23,75 +22,143 @@ def wrapped(self, *args, **kwargs):
 class InvalidSyncToken(Exception):
     pass
 
+# In Outlook, an event can be:
+# - a 'singleInstance' event,
+# - a 'serie master' which contains all the information about an event reccurrence such as
+# - an 'occurrence' which is an event from a reccurrence (serie) that follows this reccurrence
+# - an 'exception' which is an event from a reccurrence (serie) but some differences with the reccurrence template (could be
+#   the name, the day of occurrence, ...)
+#
+#  All these kinds of events are identified by:
+#  - a event ID (id) which is specific to an Outlook calendar.
+#  - a global event ID (iCalUId) which is common to all Outlook calendars containing this event.
+#
+#  - 'singleInstance' and 'serie master' events are retrieved through the end-point `/v1.0/me/calendarView/delta` which provides
+#  the last modified/deleted items since the last sync (or all of these items at the first time).
+#  - 'occurrence' and 'exception' events are retrieved through the end-point `/v1.0/me/events/{serieMaster.id}/instances`,
+#  using the corresponding serie master ID.
+
 class MicrosoftCalendarService():
 
     def __init__(self, microsoft_service):
         self.microsoft_service = microsoft_service
 
     @requires_auth_token
-    def get_events(self, sync_token=None, token=None, timeout=TIMEOUT):
+    def _get_events_delta(self, sync_token=None, token=None, timeout=TIMEOUT):
+        """
+        Get a set of events that have been added, deleted or updated in a time range.
+        See: https://docs.microsoft.com/en-us/graph/api/event-delta?view=graph-rest-1.0&tabs=http
+        """
         url = "/v1.0/me/calendarView/delta"
         headers = {'Content-type': 'application/json', 'Authorization': 'Bearer %s' % token}
         params = {}
+
         if sync_token:
             params['$deltatoken'] = sync_token
         else:
             params['startDateTime'] = '2016-12-01T00:00:00Z'
             params['endDateTime'] = '2030-1-01T00:00:00Z'
+
         try:
-            status, data, time = self.microsoft_service._do_request(url, params, headers, method='GET', timeout=timeout)
+            _, data, _ = self.microsoft_service._do_request(url, params, headers, method='GET', timeout=timeout)
```

Comment:
You should use dummy to avoid override _ if it is later used in this file.

Reply:
ok :+1: